agent/tools/toolsdir.go

```diff
@@ -216,3 +235,46 @@ func ChangeAgentTools(dataDir string, agentName string, vers version.Binary) (*c
 	}
 	return tools, nil
 }
+
+// maybeReadTools will return tools based on the dataDir and current juju version.
+// If no agent is found based on the current host series, search for agent binaries
+// for different series and the same version/arch.
+//
+// TODO: (hml) 2018-02-22
+// Added for lp:bug1749201 - once agent binaries without the series in the name are
+// available, ReadTools() can move back to ChangeAgentTools()
+func maybeReadTools(dataDir string, currentVers version.Binary) (tools *coretools.Tools, err error) {
+	defer func() {
+		err = errors.Annotate(err, "cannot read agent binaries of current or other series")
+	}()
+
+	tools, err = ReadTools(dataDir, currentVers)
+	if err == nil {
+		return tools, nil
+	}
+	logger.Tracef("%s not found (%s)", SharedToolsDir(dataDir, currentVers), err)
+
+	hostOS := utilsseries.MustOSFromSeries(utilsseries.MustHostSeries())
+	potentialSeries := utilsseries.OSSupportedSeries(hostOS)
+
+	potentialVers := currentVers
+	for _, series := range potentialSeries {
+		potentialVers.Series = series
+		tools, err := ReadTools(dataDir, potentialVers)
+		if err == nil {
```

Comment:
swap this, saves unnecessary indentation
if err != nil {
  logger......
  continue
}
.. do the rest

Reply:
done